channel.go

```diff
@@ -546,13 +532,8 @@ func (ch *Channel) addConnection(c *Connection, direction connectionDirection) b
 	}
 
 	switch state := ch.mutable.state; state {
```

Comment:
I tend to use `switch` if there's more than one conditional, I find it easier to read. The `if` doesn't clearly show "these 2 states are fine, everything else is bad" without a lot more parsing